### PR TITLE
add Leak Detector

### PR DESCRIPTION
**Pull request by luc-ass**
Hi @Danielhiversen,

I have added the new [Leak Detector](https://eu.switch-bot.com/products/switchbot-water-leak-detector) to pySwitchbot, as discussed in #246. From what I can see this should work but I am not quite sure on how to robustly test, whether all works as expected. It would be great, if you could point me in the right direction.

Thanks,
Lucas

### CODE REVIEW

File: switchbot/adv_parsers/leak.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+def process_leak(data: bytes | None, mfr_data: bytes | None) -> dict[str, bool | int]:
+    """Process SwitchBot Water Leak Detector advertisement data."""
+    if data is None and mfr_data is None:
+        return {}
+
+    water_leak_detected = None
```
Comment (bdraco) [inline switchbot/adv_parsers/leak.py:6]: Please add tests for this parser

**Comment by bdraco** [comment]
If you run 
`pre-commit install` 
`pre-commit run --all-files`

It should fix the lint issues for you

**Review by bdraco — CHANGES_REQUESTED** [review]
Please add tests for this parser

File: tests/test_adv_parser.py [hunk 2]
```diff
@@ -1737,3 +1737,33 @@ def test_parse_advertisement_data_relay_switch_1():
...
+
+
+def test_parse_advertisement_data_leak():
+    """Test parse_advertisement_data for the leak."""
+    ble_device = generate_ble_device("aa:bb:cc:dd:ee:ff", "any")
+    adv_data = generate_advertisement_data(
+        manufacturer_data={2409: b"\xc4407Lz\x18N\x98g^\x94Q<\x05\x00\x00\x00\x00"},
+        service_data={"0000fd3d-0000-1000-8000-00805f9b34fb": b"&\x00N"},
+        rssi=-72,
+    )
```
Comment (bdraco) [inline tests/test_adv_parser.py:1749]: Please add a test for passive scans as well.  In passive mode, the service_data will be empty

**Comment by bdraco** [comment]
> Hi @bdraco thank you for your input! This addition still needs to be tested with my hardware. Have you got a process how to check which response corresponds to which state? I am not super expericened with hardware testing ☺️

Try to get the device to end up in all possible states and write tests to each state to assert that the data reflects the state you believe the device to be in.  When there aren't docs its usually hours of trial and error to get it as close to correct as possible.

**Comment by luc-ass** [comment]
> Try to get the device to end up in all possible states and write tests to each state to assert that the data reflects the state you believe the device to be in. When there aren't docs its usually hours of trial and error to get it as close to correct as possible.

I have now acquired data for leak/no leak and different battery levels. I have no idea how to "tamper" with the device, but that kind of message is supported [here](https://github.com/OpenWonderLabs/node-switchbot/blob/latest/src/device/woleak.ts), so I implemented. For now I was able to test with a battery at 9% but still no low_battery bit, while the app already shows it. Looks like it is not really used.

File: switchbot/adv_parsers/leak.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    device_tampered = None
+    battery_level = None
+    low_battery = None
+
+    # Check model ID (first byte should be 0x26 for Water Leak Detector)
+    if data[0] != 0x26:
+        return {}
```
Comment (bdraco) [inline switchbot/adv_parsers/leak.py:16]: Please add coverage for this branch

File: switchbot/devices/leak.py [hunk 4]
```diff
@@ -0,0 +1 @@
+from __future__ import annotations
```
Comment (bdraco) [inline switchbot/devices/leak.py:1]: We can drop this file since there is no active connection

**Review by bdraco — APPROVED** [review]

**Comment by bdraco** [comment]
Thanks @luc-ass

**Pull request merged**